src/OrbitGgp/ClientTest.cpp

```diff
@@ -496,4 +496,86 @@ TEST_F(OrbitGgpClientTest, GetAccountAsyncWorking) {
   EXPECT_TRUE(future_is_resolved);
 }
 
+TEST_F(OrbitGgpClientTest, GetSymbolDownloadInfosAsyncWorking) {
+  auto client = CreateClient(QString::fromStdString(mock_ggp_working_.string()));
+  ASSERT_THAT(client, HasValue());
+
+  bool future_is_resolved = false;
+
+  std::vector<std::pair<std::string, std::string>> module_names_and_build_ids = {
+      {"symbol_filename_0", "build_id_0"}, {"symbol_filename_2", "build_id_2"}};
+  auto future = client.value()->GetSymbolDownloadInfoAsync(module_names_and_build_ids);
+  future.Then(
+      main_thread_executor_.get(),
+      [&future_is_resolved](const ErrorMessageOr<std::vector<SymbolDownloadInfo>>& symbols) {
+        EXPECT_FALSE(future_is_resolved);
+        future_is_resolved = true;
+        ASSERT_THAT(symbols, HasValue());
+        EXPECT_EQ(symbols.value().size(), 1);
+        EXPECT_EQ(symbols.value()[0].file_id, "symbolFiles/build_id_0/symbol_filename_0");
+        QCoreApplication::exit();
+      });
+
+  QCoreApplication::exec();
+
+  EXPECT_TRUE(future_is_resolved);
+}
+
+TEST_F(OrbitGgpClientTest, GetSymbolDownloadInfoAsyncTimeout) {
+  // mock_ggp_working_ has a 50ms sleep, hence waiting for only 5ms should result in a timeout
+  auto client = CreateClient(QString::fromStdString(mock_ggp_working_.string()),
+                             std::chrono::milliseconds{5});
+  ASSERT_THAT(client, HasValue());
+
+  bool future_is_resolved = false;
```

Comment:
I don't think you will need this bool. You are basically testing two things here:

1. You verify that the continuation is only called once, but this is ensured by the Future-framework. You are basically testing a feature from Futures here.
2. You verify that the continuation is at least called once, but this is also already tested since you call QCoreApplication::exit from the lambda expression. If it wasn't called, the exec call wouldn't finish.

Reply:
Thanks for the explanation! I didn't realize this before.

I removed this bool from the newly added tests. Will remove it from other tests in ClientTest.cpp also.